aws_lambda_builders/workflows/rust_cargo/DESIGN.md

```diff
@@ -0,0 +1,44 @@
+# Rust Cargo Builder
+
+## Scope
+
+This package enables the creation of a Lambda deployment package for Rust projects managed using the [cargo](https://doc.rust-lang.org/cargo/) build tool targeting Lambda's "provided" runtime. Rust support for the provided runtime is bundled as a compilation dependency of these projects, provided by the [lambda](https://github.com/awslabs/aws-lambda-rust-runtime) crate.
+
+## Implementation
+
+The general algorithm for preparing a rust executable for use on AWS Lambda
+is as follows.
+
+### Build
+
+It builds a binary in the standard cargo target directory. 
+
+### Copy and Rename executable
+
+It then copies the executable to the target directory renaming the executable to "bootstrap" to honor the provided runtime's [expectation on executable names](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html).
```

Comment:
@softprops I currently have the following `serverless.yml` that I would like to migrate to this SAM approach:

https://github.com/brainstorm/htsget-aws/blob/master/serverless.yml#L29

Before I do, would it be possible to have both `reads` and `variants` functions deployed and not just a single `bootstrap`?

Reply:
replied here https://github.com/awslabs/aws-lambda-builders/pull/174#issuecomment-635534361